sirepo/raydata/scan_monitor.py

```diff
@@ -157,6 +158,25 @@ def statuses_for_scans(cls, catalog_name, uids):
         )
 
 
+class _Beamline(abc.ABC):
```

Comment:
Let's create a new folder under raydata and move the classes into it each in their own file. I'm struggling to come up with a good name for the dir. Maybe analysis_driver? I don't love the name beamline because it ties us to the concept that these are for beamlines when really they are just some thing that does analysis in a specific way.

Under analysis_driver there will be an:
`__init__.py` - this will define class AnalysisDriver (currently called _Beamline)
chx.py - defines class CHX
csx.py - defines class CSX

In addition, I don't think this need to inherit from ABC. The class will possibly have methods that do things but are general to all analysis_drivers. I would just inherit from PKDict.

Reply:
the classes CHX, CSX need to be in the same file as the base class due to circular imports. Do you think it still makes sense to have this directory analysis_driver?

Reply:
We can import the child classes on initialization which gets around the circular dep problem. See how we solve it in Sirepo for job_driver's which work in the same way https://github.com/radiasoft/sirepo/blob/8fcc5e530387c9b31e4d0f6206ff3816dc339425/sirepo/job_driver/__init__.py#L402

Reply:
You could either do like we have in Sirepo (import the module and then call an init function on it). Or, you could lazily import the driver child classes on demand (ex if a request comes in for chx, check if we've imported chx, if not then import, save it in some global datastructure, use the class). I think lazy import on demand might be simpler for this case.

Reply:
141c42f960527ab2010ef7b488fcfeca76f11417